Initial import of Postgresql system role (#1)
linux-system-roles/postgresql: README.md

# PostgreSQL system role
![CI Testing](https://github.com/linux-system-roles/postgresql/workflows/tox/badge.svg)

This role installs, configures, and starts PostgreSQL Server.

The role also optimizes the datadase server settings to improve performance.

The role currently works with PostgreSQL server 10 12 and 13.
## Role Variables
### postgresql_verison
Allow set up version of Postgresql server. This role supports Postgresql 10 12 and 13
```yaml
postgresql_version: "13"
```
### postgresql_password

> Optionally you can be set up password for database super user `postgres` by default

Reviewer: ```suggestion
Optionally, you can set up password for database super user `postgres` by default.
```

> there is not a passwordi, datababase is accesible from `postgres` system account via UNIX socket.

Reviewer: ```suggestion
If there is not a password, database is accessible from `postgres` system account via UNIX socket.
```

```yaml

> postgresql_password: "mysecretpassword"

Reviewer: ```suggestion
postgresql_password: !vault |
          $ANSIBLE_VAULT;1.2;AES256;dev
          ....
```

you can also use certificate system role. For more detail see [`examples/`](examples).
```yaml
postgresql_cert_name: "server"
```
### postgresql_key_path
Optionaly you can specify path to server key using `postgresql_key_path` variable. The default value is
```yaml
postgresql_key_path: "/etc/pki/tls/private"
```
### postgresql_cert_path
Optionaly you can specify path to server cert using `postgresql_cert_path` variable. The default value is
```ymal
postgresql_cert_path: "/etc/pki/tls/certs"
```
### postgresql_input_file
For running SQL script define path to your SQL file using `postgresql_input_file`:
```yaml
postgresql_input_file: "/tmp/mypath/file.sql"
```
### postgresql_server_tuning
By default the system role makes server settings tuning based on system resources,
This functionality is enabled by default. For disabling it there is a possibility to
set up the `postgresql_server_tuning` variable.
```yaml
postgresql_server_tuning: false
```

More about usage could be found in [`examples/`](examples) directory

## Example Playbook

```yaml
- hosts: all
  vars:
    postgresql_version: "13"

>     postgresql_password: "passwd"

Reviewer: ```suggestion
    postgresql_password: !vault |
          $ANSIBLE_VAULT;1.2;AES256;dev
          ....
```

# PostgreSQL system role
![CI Testing](https://github.com/linux-system-roles/postgresql/workflows/tox/badge.svg)

This role installs, configures, and starts PostgreSQL Server.

The role also optimizes the datadase server settings to improve performance.

> The role currently works with PostgreSQL server 10 12 and 13.

Reviewer 2: ```suggestion
The role currently works with PostgreSQL server 10, 12, and 13.
```

## Role Variables
### postgresql_verison

> Allow set up version of Postgresql server. This role supports Postgresql 10 12 and 13

Reviewer 2: ```suggestion
Allow set up version of Postgresql server. This role supports Postgresql 10, 12, and 13
```

```yaml
postgresql_version: "13"
```
### postgresql_password

> Optionally, you can set up password for database super user `postgres` by default

Reviewer 2: ```suggestion
Optionally, you can set up password for database super user `postgres`. By default
```

Usage of `postgresql_pg_hba_conf` causes replacement of default upstream configuration
```yaml
postgresql_pg_hba_conf:
  - type: local
    database: all
    user: all
    auth_method: peer
  - type: host
    database: all
    user: all
    address: '127.0.0.1/32'
    auth_method: ident
  - type: host
    database: all
    user: all
    address: '::1/128'
    auth_method: ident
```
### postgresql_server_conf
Usage of `postgresql_server_conf` adds defined values at the end of postgresql.conf.
So the default ones are overwritten.
```yaml
postgresql_server_conf:
  ssl: on
  shared_buffers: 128 MB
  huge_pages: try
```
### postgresql_ssl_enable
To set up ssl connection it's necessary to set up `postgresql_ssl_enable` variable and provide server certificate and key.
```yaml
postgresql_ssl_enable: yes
```
### postgresql_cert_name
To specify certificate name use `postgresql_cert_name` variable.
You can copy your certificate to `/etc/pki/tls/certs/server.crt` and key to `/etc/pki/tls/private/server.key` or
you can also use certificate system role. For more detail see [`examples/`](examples).
```yaml

> postgresql_cert_name: "server"

Reviewer 2: ```suggestion
postgresql_cert_name: server
```
Strings do not require quotes, please remove everywhere

```
### postgresql_key_path
Optionaly you can specify path to server key using `postgresql_key_path` variable. The default value is
```yaml

> postgresql_key_path: "/etc/pki/tls/private"

Reviewer 2: ```suggestion
postgresql_key_path: /etc/pki/tls/private
```

The role also optimizes the datadase server settings to improve performance.

The role currently works with PostgreSQL server 10 12 and 13.
## Role Variables
### postgresql\_verison
Allow set up version of Postgresql server. This role supports Postgresql 10 12 and 13
```yaml
postgresql_version: "13"
```
### postgresql\_password
Set up default password for database super user `postgres`
```yaml
postgresql_password: "mysecretpassword"
```
### pg\_hba\_conf
A description of input variables that are not reqiured. Upstream configuration is used by default.
Usage of `pg_hba_conf` causes replacement of default upstream configuration
```yaml
pg_hba_conf:
  - type: local
    database: all
    user: all
    auth_method: peer
  - type: host
    database: all
    user: all
    address: '127.0.0.1/32'
    auth_method: ident
  - type: host
    database: all
    user: all
    address: '::1/128'
    auth_method: ident
```
### postgresql\_server\_conf
Usage of `postgresql_server_conf` adds defined values at the end of postgresql.conf.
So the default ones are overwritten.
```yaml
postgresql_server_conf:
  ssl: on
  shared_buffers: 128 MB
  huge_pages: try
```

> ### ssl\_enable

Reviewer 2: add the postgresql_ prefix

To set up ssl connection it's necessary to set up `ssl_enable` variable and provide server certificate and key.
```yaml
ssl_enable: yes
```

> ### cert\_name

Reviewer 2: add the postgresql_ prefix
It would be nice to support cert residing both locally and remotely. Like what `remote_src` var in the copy module does. So that customers have more flexibility.

Reviewer: Right - see https://linux-system-roles.github.io/documentation/tls_crypto_parameter_and_key_names.html for our conventions about cert/key naming

# PostgreSQL system role

> ![CI Testing](https://github.com/linux-system-roles/template/workflows/tox/badge.svg)

Reviewer: postgresql in the link also

This role installs, configures, and starts PostgreSQL Server.

> The role also optimizes the datadase server settings to improve performance.

Reviewer: database

> The role currently works with PostgreSQL server 10 12 and 13.

Reviewer: Would prefer not to have a link to something only available to Red Hat customers (unless it is publicly available)

## Role Variables
### postgresql_verison
Allow set up version of Postgresql server. This role supports Postgresql 10 12 and 13
```yaml
postgresql_version: "13"
```
### postgresql_password
Set up default password for database super user `postgres`
```yaml

> postgresql_password: "mysecretpassword"

Reviewer: Recommend using Ansible Vault to encrypt the value. See https://docs.ansible.com/ansible/latest/user_guide/vault.html

```
### postgresql_server_conf
Usage of `postgresql_server_conf` adds defined values at the end of postgresql.conf.
So the default ones are overwritten.
```yaml
postgresql_server_conf:
  ssl: on
  shared_buffers: 128 MB
  huge_pages: try
```
### postgresql_ssl_enable
To set up ssl connection it's necessary to set up `postgresql_ssl_enable` variable and provide server certificate and key.
```yaml
postgresql_ssl_enable: yes
```
### postgresql_cert_name
To specify certificate name use `postgresql_cert_name` variable.
You can copy your certificate to `/etc/pki/tls/certs/server.crt` and key to `/etc/pki/tls/private/server.key` or
you can also use certificate system role. For more detail see [`examples/`](examples).
```yaml
postgresql_cert_name: "server"
```
### postgresql_input_file
For running SQL script define path to your SQL file using `postgresql_input_file`:
```yaml
postgresql_input_file: "/tmp/mypath/file.sql"
```
### postgresql_server_tuning
By default the system role makes server settings tuning based on system resources,
This functionality is enabled by default. For disabling it there is a possibility to
set up the `postgresql_server_tuning` variable.
```yaml

> postgresql_server_tuning: false

Reviewer: There is a `tuned` profile for postgresql https://github.com/redhat-performance/tuned/tree/master/profiles/postgresql - do we want to support this in the role?  If so, the `mssql` role does something similar https://github.com/linux-system-roles/mssql/blob/master/tasks/main.yml#L219-L270
